internal/command/launch/launch.go

```diff
@@ -82,10 +84,15 @@ func (state *launchState) Launch(ctx context.Context) error {
 	}
 
 	// Finally write application configuration to fly.toml
-	configPath := state.configPath
-	if flag.GetBool(ctx, "json") {
+	configDir := filepath.Dir(state.configPath)
```

Comment:
Minor suggestion: 

```go
configDir, configFile := filepath.Split(state.configPath)
```

And then later:

```go
configPath = filepath.Join(configPath, configFile)
```

The concern is that at some point in the future, the default format may change, and the person making that change may miss the `fly launch --config dirname` path...

Reply:
I think I get what you mean, but I'm not 100% sure :sweat_smile: 

I've pushed an updated version that is a bit cleaner and I think resolves the issue you describe; close the issue if this is the case and otherwise maybe you can give me an example of where this might fail :) 

thanks for the :eyes: !